subiquity/server/controllers/filesystem.py

```diff
@@ -1293,6 +1292,7 @@ async def _probe(self, *, context=None):
             finally:
                 elapsed = time.time() - start
                 log.debug(f"{short_label} probing took {elapsed:.1f} seconds")
+            self.start_monitor()
```

Comment:
If there is a probing operation running when storage is marked configured, won't this call restart the monitoring despite the call to `self.stop_monitor()` in `configured()` ?

Reply:
Also, this call will not be executed in case in exception is raised, it this intended?

Reply:
Maybe a `add_done_callback(self.start_monitor)` would be more suitable. And that would leave the option to call `remove_done_callback(self.start_monitor)` in the `configured` method.

Reply:
It's intentional to not monitor on the exception case, because today we have no plan forward after the probing failure due to timeout.  Perhaps if probing timeout failure were recoverable.

Also the restricted probe is going to go away, so either our probe is passing and everything is great or we have no interest in listening for more events.

Reply:
> add_done_callback

An interesting idea, but this would be called on timeout, which I don't think we want.

Reply:
> It's intentional to not monitor on the exception case, because today we have no plan forward after the probing failure due to timeout. Perhaps if probing timeout failure were recoverable.
> 
> Also the restricted probe is going to go away, so either our probe is passing and everything is great or we have no interest in listening for more events.

So you're suggesting that we don't do retries in case of probing failure? I'm not 100% convinced that we should consider retries useless. If you hit a timeout and then eject a device (that has multiple partitions?) for instance, there is a chance that you will get a successful probing the second time. But if you think the change does more good than bad as is, then it's probably acceptable.

> If there is a probing operation running when storage is marked configured, won't this call restart the monitoring despite the call to self.stop_monitor() in configured() ?

I still think this is something we want to fix though. Thoughts ?

Reply:
> If you hit a timeout and then eject a device (that has multiple partitions?) for instance, there is a chance that you will get a successful probing the second time.

This sounds plausible.  I'm moving start_monitor() to the `finally` block.  Nice suggestion.